main.js

```diff
@@ -176,6 +183,32 @@ if (buildStatic) {
     return;
 }
 
+function getFileHeaders(headerPath) {
+    if (headerPath === 'NO_HEADERS_FILE') {
+        return {};
+    }
+
+    try {
+        return JSON.parse(fs.readFileSync(`${headerPath}`));
+    } catch (error) {
+        console.error(`Error parsing JSON for headers file ${headerPath}`);
```

Comment:
Should we stop the server from starting at this point? I think it's better to fail hard here, otherwise the user might be extremely confused as to why their headers aren't being set. Let's discuss